### PR TITLE
#12224: Add optional output tensor support for lt_bw

### LINKED ISSUE #12224: Add queue_id and optional output tensors to ttnn.lt

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #12224 

Work Done : Overloaded Unary and Binary LT with optional output tensor and Q_ID support
[CI](https://github.com/tenstorrent/tt-metal/actions/runs/10899411038) : Passed

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 1]
```diff
@@ -556,6 +556,37 @@ std::vector<Tensor> ExecuteBackwardBiasGelu::invoke(
...
+
+std::vector<std::optional<Tensor>> ExecuteBackwardLt::invoke(
+    uint8_t queue_id, const Tensor& grad, const Tensor& input, float other, const std::optional<MemoryConfig>& output_mem_config,
+    std::optional<Tensor> input_grad) {
+    std::vector<std::optional<Tensor>> result;
+    if(!input_grad.has_value()){
+        input_grad = input;
+    }
+    ttnn::zeros_like(queue_id, grad, grad.get_dtype(), grad.get_layout(), std::nullopt, output_mem_config, input_grad);
+    result.push_back(input_grad.value());
```
Comment (KalaivaniMCW) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:586]: instead, 

try, 
result[0] = input_grad.has_value() ? ttnn::zeros_like(queue_id, grad, grad.get_dtype(), grad.get_layout(), std::nullopt, output_mem_config, input_grad) :  ttnn::zeros_like(queue_id, grad, grad.get_dtype(), grad.get_layout(), std::nullopt, output_mem_config)

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward.hpp [hunk 2]
```diff
@@ -144,6 +144,26 @@ struct ExecuteBackwardBiasGelu {
...
+        std::optional<Tensor> input_grad = std::nullopt,
+        std::optional<Tensor> other_grad = std::nullopt);
+
+    static std::vector<std::optional<Tensor>> invoke(
+        uint8_t queue_id,
+        const Tensor &grad_tensor_arg,
+        const Tensor &input_tensor_arg,
+        float other,
+        const std::optional<MemoryConfig> &memory_config = std::nullopt,
+        std::optional<Tensor> input_grad = std::nullopt);
```
Comment (KalaivaniMCW) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/binary_backward.hpp:164]: we need to have the same invoke methods without queue id as well

**Review by KalaivaniMCW — APPROVED** [review]
Approved.

**Review by eyonland — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10915611277 - passed

**Pull request merged**